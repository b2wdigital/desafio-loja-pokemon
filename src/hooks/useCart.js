import { useState } from 'react'

export const useCart = () => {
  const [open, setOpen] = useState(false)

  const toggleDrawer = () => {
    console.log('toggleDrawer', !open)
    setOpen(status => !status)
  }

  return {
    open,
    toggleDrawer
  }
}
