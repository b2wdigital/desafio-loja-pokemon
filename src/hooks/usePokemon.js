import { useState, useCallback } from 'react'
import PokeApiService from '../services/pokeapi'

export const usePokemon = () => {
  const [loading, setLoading] = useState(false)
  const [error, setError] = useState(false)
  const [data, setData] = useState([])

  const getPokemonDetail = async (pokemons) => {
    const pokemonListDetails = await Promise.all(
      pokemons.map(async ({ url }) => {
        try {
          const responseDetails = await PokeApiService.getDetails(url)
          return responseDetails && responseDetails.data
        } catch (exception) {
          return {}
        }
      })
    )

    return pokemonListDetails
  }

  const getAllPokemon = useCallback(async () => {
    try {
      setError(false)
      setLoading(true)

      const responseList = await PokeApiService.getAll(0, 60)
      let responseDetails = await getPokemonDetail(responseList.data.results)
      responseDetails = responseDetails.filter(value => Object.keys(value).length !== 0)

      setData(responseDetails)
    } catch (err) {
      if (err.response && err.response.status === 404) {
        setData([])
      } else {
        setError(true)
      }
    } finally {
      setLoading(false)
    }
  }, [])

  return {
    loading,
    data,
    error,
    getAllPokemon
  }
}
