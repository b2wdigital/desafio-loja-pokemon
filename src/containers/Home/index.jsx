/* eslint-disable no-unused-expressions */

import React, { useEffect } from 'react'
import Alert from '@material-ui/lab/Alert'
import { CardActionArea, Container, Grid } from '@material-ui/core'
import Loading from '../../components/Loading'
import { usePokemon } from '../../hooks/usePokemon'
import { GridStyled, CardStyled, ImageStyled, CardContentStyled, NameStyled, TypesStyled, PriceStyled, AvatarStyled } from './style'

const Home = () => {
  const {
    loading,
    data,
    error,
    getAllPokemon
  } = usePokemon()

  useEffect(() => {
    getAllPokemon()
  }, [getAllPokemon])

  useEffect(() => {
    error &&
      <Alert variant="filled" severity="error">
        Ocorreu um erro, tente novamente!
      </Alert>
  }, [error])

  return (
    <Container>
      {loading && <Loading backdrop={true} /> }
      <Grid container spacing={2}>
        {data.length > 0 && data.map(item => (
          <GridStyled item xs={2} key={item.id}>
            <CardStyled>
              <CardActionArea>
                <ImageStyled
                  alt={item.name}
                  title={item.name}
                  src={item.sprites.front_default}
                  onMouseOver={(e) => {
                    e.currentTarget.src = item.sprites.back_default
                  }}
                  onMouseOut={(e) => {
                    e.currentTarget.src = item.sprites.front_default
                  }}
                />
                <CardContentStyled>
                  <NameStyled variant="h5">
                    {item.name}
                  </NameStyled>
                  <PriceStyled variant="subtitle1">
                    <b>$</b> {item.base_experience}
                  </PriceStyled>
                  <TypesStyled>
                    {item.types.map(type => (
                      <AvatarStyled
                        key={type.type.name}
                        src={require(`../../assets/icons/${type.type.name}.svg`).default}
                        alt={type.type.name}
                      />
                    ))}
                  </TypesStyled>
                </CardContentStyled>
              </CardActionArea>
            </CardStyled>
          </GridStyled>
        ))}
      </Grid>
    </Container>
  )
}

export default Home
