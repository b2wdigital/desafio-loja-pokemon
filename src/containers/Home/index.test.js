import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { MuiThemeProvider } from '@material-ui/core'
import { ThemeProvider } from 'styled-components'
import Home from '.'
import mockDetailsPokemon from '../../assets/mock-data/details-pokemon.json'
import theme from '../../theme'
import { usePokemon } from '../../hooks/usePokemon'

jest.mock('../../hooks/usePokemon')

const containerComponent = (
  <MuiThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </ThemeProvider>
  </MuiThemeProvider>
)

describe('Home container', () => {
  test('Render layout', () => {
    const mockHooks = {
      loading: true,
      data: [mockDetailsPokemon],
      error: false,
      setError: jest.fn(),
      offset: 0,
      paginate: jest.fn(),
      getAllPokemon: jest.fn()
    }
    usePokemon.mockReturnValue(mockHooks)

    const component = render(containerComponent)

    fireEvent.mouseOver(screen.getByAltText('bulbasaur'))
    fireEvent.mouseOut(screen.getByAltText('bulbasaur'))

    expect(component).toMatchSnapshot()
  })

  test('Render without pokemon', () => {
    const mockHooks = {
      loading: false,
      data: [],
      error: false,
      setError: jest.fn(),
      offset: 0,
      paginate: jest.fn(),
      getAllPokemon: jest.fn()
    }
    usePokemon.mockReturnValue(mockHooks)

    const component = render(containerComponent)
    expect(component).toMatchSnapshot()
  })

  test('Render with error', () => {
    const mockHooks = {
      loading: false,
      data: [],
      error: true,
      setError: jest.fn(),
      offset: 0,
      paginate: jest.fn(),
      getAllPokemon: jest.fn()
    }
    usePokemon.mockReturnValue(mockHooks)

    const component = render(containerComponent)

    expect(component).toMatchSnapshot()
  })
})
