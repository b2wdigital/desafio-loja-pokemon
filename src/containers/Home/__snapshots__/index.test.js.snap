// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container Render layout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-jrAGrp fncGdr MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <img
            class="sc-kEjbxe fRIJKq"
            src="logo.png"
          />
          <div
            class="sc-iqHYGH dhxBkJ"
          >
            <div
              class="sc-crrsfI VNyzD"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root sc-dQppl iankGQ"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input"
                placeholder="Procurar..."
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root sc-bdfBwQ iodWSc"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="sc-gsTCUz gomrDe"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root sc-dlfnbm hbZnVW MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2"
          >
            <div
              class="MuiPaper-root MuiCard-root sc-hKgILt gJCffI MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="bulbasaur"
                  class="sc-eCssSg fFPGzd"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                  title="bulbasaur"
                />
                <div
                  class="MuiCardContent-root sc-jSgupP cuDwNq"
                >
                  <h5
                    class="MuiTypography-root sc-gKsewC hhnpme MuiTypography-h5"
                  >
                    bulbasaur
                  </h5>
                  <h6
                    class="MuiTypography-root sc-pFZIQ hZisWE MuiTypography-subtitle1"
                  >
                    <b>
                      $
                    </b>
                     
                    64
                  </h6>
                  <div
                    class="MuiBox-root MuiBox-root-1 sc-iBPRYJ kEDcat"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                    >
                      <img
                        alt="grass"
                        class="MuiAvatar-img"
                        src="grass.svg"
                      />
                    </div>
                    <div
                      class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                    >
                      <img
                        alt="poison"
                        class="MuiAvatar-img"
                        src="poison.svg"
                      />
                    </div>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
              data-testid="btnLoadMore"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Carregar mais
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-jrAGrp fncGdr MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <img
          class="sc-kEjbxe fRIJKq"
          src="logo.png"
        />
        <div
          class="sc-iqHYGH dhxBkJ"
        >
          <div
            class="sc-crrsfI VNyzD"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root sc-dQppl iankGQ"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input"
              placeholder="Procurar..."
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root sc-bdfBwQ iodWSc"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="sc-gsTCUz gomrDe"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root sc-dlfnbm hbZnVW MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2"
        >
          <div
            class="MuiPaper-root MuiCard-root sc-hKgILt gJCffI MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="bulbasaur"
                class="sc-eCssSg fFPGzd"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                title="bulbasaur"
              />
              <div
                class="MuiCardContent-root sc-jSgupP cuDwNq"
              >
                <h5
                  class="MuiTypography-root sc-gKsewC hhnpme MuiTypography-h5"
                >
                  bulbasaur
                </h5>
                <h6
                  class="MuiTypography-root sc-pFZIQ hZisWE MuiTypography-subtitle1"
                >
                  <b>
                    $
                  </b>
                   
                  64
                </h6>
                <div
                  class="MuiBox-root MuiBox-root-1 sc-iBPRYJ kEDcat"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                  >
                    <img
                      alt="grass"
                      class="MuiAvatar-img"
                      src="grass.svg"
                    />
                  </div>
                  <div
                    class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                  >
                    <img
                      alt="poison"
                      class="MuiAvatar-img"
                      src="poison.svg"
                    />
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          align="center"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
            data-testid="btnLoadMore"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Carregar mais
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home container Render with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-jrAGrp fncGdr MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <img
            class="sc-kEjbxe fRIJKq"
            src="logo.png"
          />
          <div
            class="sc-iqHYGH dhxBkJ"
          >
            <div
              class="sc-crrsfI VNyzD"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root sc-dQppl iankGQ"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input"
                placeholder="Procurar..."
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight"
        >
          <div
            class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation0"
            direction="down"
            role="alert"
            style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message"
            >
              Ocorreu um erro, tente novamente!
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Nenhum Pokemon encontrado...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-jrAGrp fncGdr MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <img
          class="sc-kEjbxe fRIJKq"
          src="logo.png"
        />
        <div
          class="sc-iqHYGH dhxBkJ"
        >
          <div
            class="sc-crrsfI VNyzD"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root sc-dQppl iankGQ"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input"
              placeholder="Procurar..."
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight"
      >
        <div
          class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation0"
          direction="down"
          role="alert"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message"
          >
            Ocorreu um erro, tente novamente!
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          align="center"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Nenhum Pokemon encontrado...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home container Render without pokemon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-jrAGrp fncGdr MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <img
            class="sc-kEjbxe fRIJKq"
            src="logo.png"
          />
          <div
            class="sc-iqHYGH dhxBkJ"
          >
            <div
              class="sc-crrsfI VNyzD"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root sc-dQppl iankGQ"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input"
                placeholder="Procurar..."
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Nenhum Pokemon encontrado...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-jrAGrp fncGdr MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <img
          class="sc-kEjbxe fRIJKq"
          src="logo.png"
        />
        <div
          class="sc-iqHYGH dhxBkJ"
        >
          <div
            class="sc-crrsfI VNyzD"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root sc-dQppl iankGQ"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input"
              placeholder="Procurar..."
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          align="center"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Nenhum Pokemon encontrado...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
